krita_batch_exporter/Infrastructure.py

```diff
@@ -14,8 +14,46 @@
 
 
 def dataToPIL(wnode):
-    img = wnode.node.projectionPixelData(*wnode.bounds).data()
-    img = Image.frombytes("RGBA", wnode.size, img, "raw", "BGRA", 0, 1)
+    """
+    Returns an Image object using the Pillow library with 8-bit sRGB data that Pillow can then process.
+    If the input node's color space isn't 8-bit sRGB, uses Krita to convert the document to the right color space.
+    """
+
+    SRGB_PROFILE = "sRGB-elle-V2-srgbtrc.icc"
+
+    [x, y, w, h] = wnode.bounds
+
+    if wnode.node.colorModel() == "RGBA" and wnode.node.colorDepth() == "U8" and wnode.node.colorProfile() == SRGB_PROFILE:
+        pixel_data = wnode.node.projectionPixelData(x, y, w, h).data()
+        img = Image.frombytes("RGBA", wnode.size, pixel_data, "raw", "BGRA", 0, 1)
+        return img
```

Comment:
This branching is not good. We don't have a styleguide for Python right now that I can link you to, but I'd like you to change it to make it more future-proof and readable.
In general, you don't want to return values in the middle of functions, you create branching paths that are harder to track like that.

Instead you can do something like
```py
img = None
if is_srgb and wnode.node.colorModel() == "RGBA" and wnode.node.colorDepth() == "U8":
    pixel_data = wnode.node.projectionPixelData(x, y, w, h).data()
    img = Image.frombytes("RGBA", wnode.size, pixel_data, "raw", "BGRA", 0, 1)
else:
    # convert with Krita

return img

Reply:
This is unknown to me - I thought the conventional wisdom is supposed to be early returns. I have no strong feelings about it either way, though. I will change it, if that's how you want the code to be.

Reply:
Defensive returns at the start of a function generally make sense, to prevent anything from happening if some conditions aren't met. If you see code where people return data mid-function, nested conditions and other things like these, it can have more to do with people not following any guidelines or not caring too much about the code they write.

Reply:
Fair enough.